tools/src/test/java/com/example/core/broker/testkit/services/DeltaDownstreamServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.example.core.broker.testkit.services;
+
+import akka.Done;
+import akka.NotUsed;
+import akka.stream.javadsl.Flow;
+import com.lightbend.lagom.javadsl.api.ServiceCall;
+import org.pcollections.PSequence;
+import org.pcollections.TreePVector;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+public class DeltaDownstreamServiceImpl implements DeltaDownstreamService {
+
+    private final List<ReceivedMessage> receivedMessages = new ArrayList<>();
```

Comment:
🙇 
My original impl was ignoring the fact that `TreePVector.from(Collection)` uses the Collection's `iterator`.

Will actually use a mutable queue in `CharlieDownstream` (a dummy service consuming from two topics).
Will use a `volatile PSequence` in `DeltaDownstream` and `FoxtrotDownstream` (since Delta and Foxtrot only consume from one topic). But I think `volatile` is unnecessary since we accept the `Service` return eventually consistent values.